### DISCUSSION TITLE
Forcing a Unique Rule to Ignore a Given ID in a FormRequest

### DISCUSSION THREAD

**Discussion by prescriptionlifeline** [Laravel-Backpack/community-forum#1025] [Q&A (Help)] [answered]
I'm trying to make it so that CRUD validation forces a unique rule to ignore the ID of the user being edited. In the "Forcing a Unique Rule to Ignore a Given ID" section of https://laravel.com/docs/11.x/validation#rule-unique there's this:

```
Validator::make($data, [
    'email' => [
        'required',
        Rule::unique('users')->ignore($user->id),
    ],
]);
```
In https://github.com/Laravel-Backpack/community-forum/discussions/1015#discussioncomment-9724774 it was suggested I do `backpack_user()->id` but that doesn't work as that's equal to the user _currently logged in_ - not to the user being edited.

I'm using this to identify the class that's doing the validation:

```
CRUD::setValidation(OldUserRequest::class)
```
If I wasn't using `CRUD::setValidation` then maybe https://stackoverflow.com/a/66301928/569976 would work but, as is, it doesn't.

Is the only way to get this to work to use [array validation](https://backpackforlaravel.com/docs/6.x/crud-operation-create#b-validating-fields-using-a-rules-array)?

**Comment by jcastroa87** [comment, marked as answer]
Hello @prescriptionlifeline 

Did you try to use:
```
'email' => 'required|unique:users,email,' . $this->id,
```

Let me know if thats work for you.

Cheers.

**Reply by prescriptionlifeline** [reply]
That works - thanks!

**Comment by jcastroa87** [comment]
Im glad thats work for you. I will close the issue, but please feel free to re-open or create a new one if needed.

Cheers.
